File: amicable-numbers/amicable-numbers.py
##Problem 21

##Evaluate the sum of all the amicable numbers under 10000.

def properDivisors(n):
    divisors = list(filter(lambda x: (n % x == 0), [ x for x in range(1,n-1)])) 
    return divisors

numbers = [x for x in range(1,50)]

divs = list(zip(numbers, map(sum, map(properDivisors, numbers))))

number_list = [ x for x in range(1,1000) if x % 5 == 0 and x % 3 == 0]

#def amicable(rangemax=20000):
#    n2divsum = {n: sum(proper_divs(n)) for n in range(1, rangemax + 1)}
#    for num, divsum in n2divsum.items():
#        if num < divsum and divsum <= rangemax and n2divsum[divsum] == num:
#            yield num, divsum

print(divs)

File: amicable-numbers/amicable-numbers.js
//Problem 21

//Evaluate the sum of all the amicable numbers under 10000.

let rangemax = 3000

const createList = n => {
  return Array.from([...Array(n).keys()], x => x)
}

const properDivisors = n =>{
    let divs = createList(n).filter(x => n % x == 0)
    return divs
}

const sum = xs => xs.reduce((a, x) => a + x, 0);

let n2divsum= {}

for (let n = 1; n < rangemax; n++) {
   n2divsum[n] = (sum(properDivisors(n)))
}

let final = []

for (const [key, value] of Object.entries(n2divsum)) {
    if ((key < value) && (value <= rangemax) && (n2divsum[value] == key)){
      final.push([key, value])  
    }
}

console.log(final)
